client/my-sites/checkout/composite-checkout/components/jetpack-product-pricing-disclaimer.tsx

```diff
@@ -0,0 +1,34 @@
+import { JETPACK_FIRST_YEAR_50_PERCENT_OFF_PRODUCTS } from '@automattic/calypso-products';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { useTranslate } from 'i18n-calypso';
+import CheckoutTermsItem from 'calypso/my-sites/checkout/composite-checkout/components/checkout-terms-item';
+import useCartKey from 'calypso/my-sites/checkout/use-cart-key';
+
+export default function JetpackProductPricingDisclaimer() {
+	const cartKey = useCartKey();
+	const translate = useTranslate();
+	const { responseCart } = useShoppingCart( cartKey );
+
+	const products = responseCart?.products;
+	const product_slugs = products.map( ( product ) => product.product_slug );
+	if ( product_slugs.length === 0 ) {
+		return null;
+	}
+	const product_names = products.map( ( product ) => product.product_name );
+
+	const showJetpackProductPricingDisclaimer = JETPACK_FIRST_YEAR_50_PERCENT_OFF_PRODUCTS.filter(
+		( slug ) => product_slugs.includes( slug )
+	);
+	if ( showJetpackProductPricingDisclaimer.length > 0 ) {
+		return (
+			<CheckoutTermsItem key="jetpack-product-pricing-disclaimer">
+				{ translate(
+					'The following products are currently in development: %(productNames)s Enjoy using them at half price over the next year while we continue to develop the features.',
```

Comment:
Do you want a period between the list of product names and the next sentence?
```suggestion
					'The following products are currently in development: %(productNames)s. Enjoy using them at half price over the next year while we continue to develop the features.',
```

Reply:
It looks [like](p1675291379826429/1675244371.316719-slack-C02JJ910CNL
) we are doing this only for Jetpack Social Advanced, so I have updated the text and the code to reflect that.